mod-source-record-manager-server/src/main/java/org/folio/dao/JournalRecordDaoImpl.java

```diff
@@ -360,31 +365,44 @@ private RecordProcessingLogDto mapRowSetToRecordProcessingLogDto(RowSet<Row> res
     if (resultSet.size() == 0) {
       throw new NotFoundException("Can`t find record with specific jobExecutionId and recordId");
     }
-    resultSet.forEach(row ->
-      recordProcessingLogSummary
-        .withJobExecutionId(row.getValue(JOB_EXECUTION_ID).toString())
-        .withSourceRecordId(row.getValue(SOURCE_ID).toString())
-        .withSourceRecordOrder(row.getInteger(SOURCE_RECORD_ORDER))
-        .withSourceRecordTitle(row.getString(TITLE))
-        .withSourceRecordActionStatus(mapNameToEntityActionStatus(row.getString(SOURCE_RECORD_ACTION_STATUS)))
-        .withError(row.getString(SOURCE_ENTITY_ERROR))
-        .withRelatedInstanceInfo(constructProcessedEntityInfoBasedOnEntityType(row,
-          INSTANCE_ACTION_STATUS, INSTANCE_ENTITY_ID, INSTANCE_ENTITY_HRID, INSTANCE_ENTITY_ERROR))
-        .withRelatedHoldingsInfo(constructProcessedEntityInfoBasedOnEntityType(row,
-          HOLDINGS_ACTION_STATUS, HOLDINGS_ENTITY_ID, HOLDINGS_ENTITY_HRID, HOLDINGS_ENTITY_ERROR))
-        .withRelatedItemInfo(constructProcessedEntityInfoBasedOnEntityType(row,
-          ITEM_ACTION_STATUS, ITEM_ENTITY_ID, ITEM_ENTITY_HRID, ITEM_ENTITY_ERROR))
-        .withRelatedAuthorityInfo(constructProcessedEntityInfoBasedOnEntityType(row,
-          AUTHORITY_ACTION_STATUS, AUTHORITY_ENTITY_ID, null, AUTHORITY_ENTITY_ERROR))
-        .withRelatedPoLineInfo(new RelatedPoLineInfo()
-            .withActionStatus(mapNameToEntityActionStatus(row.getString(PO_LINE_ACTION_STATUS)))
-            .withIdList(constructListFromColumn(row, PO_LINE_ENTITY_ID))
-            .withHridList(constructListFromColumn(row, PO_LINE_ENTITY_HRID))
-            .withError(row.getString(PO_LINE_ENTITY_ERROR))
-            .withOrderId(row.getString(ORDER_ENTITY_ID)))
-        .withRelatedInvoiceInfo(constructProcessedEntityInfoBasedOnEntityType(row,
-          INVOICE_ACTION_STATUS, INVOICE_ENTITY_ID, INVOICE_ENTITY_HRID, INVOICE_ENTITY_ERROR))
-        .withRelatedInvoiceLineInfo(constructInvoiceLineInfo(row)));
+    List<ProcessedHoldingsInfo> processedHoldingsInfo = new LinkedList<>();
+    List<ProcessedItemInfo> processedItemInfo = new LinkedList<>();
+
+    Row row = resultSet.iterator().next();
+    recordProcessingLogSummary
```

Comment:
`recordProcessingLogSummary`  creation could be moved under the `resultSet` size check thus it would not be created in case of empty `resultSet`

Reply:
Thanks, done